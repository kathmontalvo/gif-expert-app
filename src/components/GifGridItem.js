import React from 'react'
import 'animate.css'

const GifGridItem = ( { id, title, url} ) => {

  return (
    <div className='card animate__zoomIn animate__animated'>
      <img src={ url} alt={title} />
      <p> { title } </p>
    </div>
  )
}

export default GifGridItem