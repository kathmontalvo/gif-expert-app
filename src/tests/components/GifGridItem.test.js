import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {
    const title = 'Dark';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe mostrar el elemento correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    })

    test('debe tener la img igual al url', () => {
        const img = wrapper.find('img');
        const src = img.prop('src');
        const alt = img.prop('alt');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('debe tener la clase animate__zoomIn ', () => {

        const div = wrapper.find('div');
        //div.props() <-- para conocer las propiedades que contiene
        const classes = div.prop('className');
        
        expect(classes.includes('animate__zoomIn')).toBe(true);
        
    })
    
    
    
    
})
