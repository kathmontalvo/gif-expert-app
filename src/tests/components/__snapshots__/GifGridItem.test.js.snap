// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar el elemento correctamente 1`] = `
<div
  className="card animate__zoomIn animate__animated"
>
  <img
    alt="Dark"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Dark
     
  </p>
</div>
`;
